zoom_recording_ingester_cdk/zoom_recording_ingester_cdk_stack.py

```diff
@@ -0,0 +1,683 @@
+                    }
+                )
+            ]
+        )
+
+        ingest = api.root.add_resource("ingest")
+        on_demand_integration = apigw.LambdaIntegration(on_demand_function)
+        ingest_method = ingest.add_method(
+            "POST",
+            on_demand_integration,
+            request_parameters={
+                "method.request.querystring.uuid": True,
+                "method.request.querystring.oc_series_id": False,
+                "method.request.querystring.allow_multiple_ingests": False
+            },
+            method_responses=[
+                apigw.MethodResponse(
+                    status_code="200",
+                    response_models={
+                        "application/json": apigw.Model.EMPTY_MODEL
+                    }
+                )
+            ]
+        )
+
+        on_demand_function.add_environment(
+            "WEBHOOK_ENDPOINT_URL",
+            (f"https://{api.rest_api_id}.execute-api.{self.region}"
+             f".amazonaws.com/{self.lambda_release_alias}/new_recording")
+        )
+
+        """
+        CloudWatch Events
+        """
+
+        download_trigger = self._create_event_rule(
+            downloader_function, "downloader", rate_in_minutes=5
+        )
+        upload_trigger = self._create_event_rule(
+            uploader_function, "uploader", rate_in_minutes=20
+        )
+
+
+        """
+        CodeBuild project
+        """
+
+        codebuild_project = codebuild.Project(
+            self, "CodeBuildProject",
+            project_name=f"{self.stack_name}-codebuild",
+            source=codebuild.Source.git_hub_enterprise(
+                https_clone_url="https://github.com/harvard-dce/zoom-recording-ingester.git",
+                clone_depth=1
+            ),
+            environment=codebuild.BuildEnvironment(
+                build_image=codebuild.LinuxBuildImage.AMAZON_LINUX_2_2,
+                compute_type=codebuild.ComputeType.SMALL
+            ),
+            artifacts=codebuild.Artifacts.s3(
+                name=self.stack_name,
+                bucket=self.lambda_code_bucket
+            ),
+            badge=True,
+            timeout=core.Duration.minutes(5)
+        )
+
+        codebuild_project.add_to_role_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "lambda:UpdateFunctionCode",
+                    "lambda:PublishVersion",
+                    "lambda:UpdateAlias"
+                ],
+                resources=[
+                    on_demand_function.function_arn,
+                    webhook_function.function_arn,
+                    downloader_function.function_arn,
+                    uploader_function.function_arn,
+                    op_counts_function.function_arn,
+                    log_notification_function.function_arn
+                ]
+            )
+        )
+
+        """
+        SNS Topic
+        """
```

Comment:
I feel like everywhere you have one of these `"""` headers could be a new class/construct